### PR TITLE
added code to createUser to check for a deleted user *before* we try …

### PR DESCRIPTION
**Pull request by BenRMitchell**
…to create.  Switches to restore the user instead if one found

I think this is the right way to create this PR for your guys' process.  Let me know if I'm wrong!

### CODE REVIEW

**Review by carnellj-genesys — APPROVED** [review]

**Review by carnellj-genesys — APPROVED** [review]
@charliecon Unless you see problems with test cases can you please review.  If it looks good please merge.

**Review by charliecon — APPROVED** [review]

File: genesyscloud/resource_genesyscloud_user.go [hunk 1]
```diff
@@ -457,6 +457,13 @@ func createUser(ctx context.Context, d *schema.ResourceData, meta interface{}) d
 		return addrErr
 	}
 
+	// Check for a deleted user before creating
+	id, _ := getDeletedUserId(email, usersAPI)
+	if id != nil {
+		d.SetId(*id)
+		return restoreDeletedUser(ctx, d, meta, usersAPI)
+	}
```
[the remark this answers is not in the document]
Comment (BenRMitchell) [reply genesyscloud/resource_genesyscloud_user.go:465]: Yep it should update correctly!  That restoreDeletedUser method updates the state of the found deleted user, then calls updateUser with that user.

[the remark this answers is not in the document]
Comment (carnellj-genesys) [reply genesyscloud/resource_genesyscloud_user.go:465]: Hey @BenRMitchell would you mind writing a test case to actually provide out the restore and delete.

[the remark this answers is not in the document]
Comment (BenRMitchell) [reply genesyscloud/resource_genesyscloud_user.go:465]: hey @carnellj-genesys, I think I've got the unit test added correctly. If it needs anything else, let me know!
Ben

**Review by carnellj-genesys — APPROVED** [review]

**Pull request merged**